content/blogposts/2023/blah.md

```diff
@@ -0,0 +1,151 @@
+    <Figure
+        src="https://storage.googleapis.com/sourcegraph-assets/blog/scip-powered-cody/req-no-scip2.png"
+        alt="'Stock Cody' hallucinating the fields of a struct."
+        caption="'Stock Cody' hallucinating the fields of a struct." />
+</a>
+
+Unfortunately, the fields listed in Cody's response aren't correct. These types of hallucinations, when trusted because they appear correct at first glance, can be particularly frustrating after aiding in building an imprecise mental model of the code that later needs to be torn down and rebuilt.
+
+The `CodyCompletionRequsetParameters` struct, in reality, contain the fields `Prompt`, `Messages`, `MaxTokensToSample`, `Temperature`, `StopSequences`, `TopK`, `TopP`, `Model` (embedded under a named struct), and `Fast`.
+
+The fields `Repo`, `File`, `Line`, `Column`, and `Prefix` given by Cody have zero overlap with this type, although we do have other request and response structs with similar sets of fields. The `Model` field is only present because it could be syntactically derived (see line 54 in the above image). The assignment to `requestParams.Model` implies its existence on the type, hence its inclusion in the response. The `TODO` comment attached to the assignment gives a local indication of its purpose, hence its rephrasing in the response.
+
+Why the discrepency? Well, 'Stock Cody' is powered by searching segments of your repository that has been embedded into a vector space for code _similar_ to the content of your question. And this works *really well* for questions pertaining to the 10,000ft view. _"How do we handle auth in service X?"_ is a question that is likely well-served by embeddings search. Code related to the implementation and handling of auth will likely make it into the LLM context window, and the high-nature level of the question enables fairly accurate responses to be generated from partial context.
+
+In contrast, our question is asking for a _highly precise_ answer related to a specific piece of code. Searching the repository's embeddings for _similar_ code is not the context needed by the LLM to make the correct inferrences. For such implementation-specific questions, we need to be able to explore a different relationship within the code graph.
+
+For the same question, we can generate a very good response from Cody by supplying additional code we find traversing the _definition_ edges of the semantically-indexed portion of the code graph. To start the search, we extract symbol names from the code that is currently in the view of the user, such as the current workspace, the user's text selection, open files, etc. These symbol names are translatted into a location in our SCIP-backed code graph. Then, particular edges are followed out to explore the graph. As we traverse the graph in this manner, we recall code that is semantically related to the code in question.
```

Comment:
"we can generate" reads a bit confusingly to me. Maybe "we can improve the response from Cody..."